slsDetectorSoftware/src/Module.cpp

```diff
@@ -1389,19 +1389,25 @@ void Module::setInjectChannel(const int offsetChannel,
     sendToDetector(F_SET_INJECT_CHANNEL, args, nullptr);
 }
 
-void Module::sendVetoPhoton(const int chipIndex, int *gainIndices,
-                            int *values) {
+void Module::sendVetoPhoton(const int chipIndex, std::vector<int> gainIndices,
+                            std::vector<int> values) {
+    const int nch = gainIndices.size();
+    if (gainIndices.size() != values.size()) {
+        throw RuntimeError("Number of Gain Indices and values do not match! "
+                           "Gain Indices size: " +
+                           std::to_string(gainIndices.size()) +
+                           ", values size: " + std::to_string(values.size()));
+    }
+    LOG(logDEBUG1) << "Sending veto photon/file to detector [chip:" << chipIndex
+                   << ", nch:" << nch << "]";
     int fnum = F_SET_VETO_PHOTON;
     int ret = FAIL;
-    int nch = shm()->nChan.x;
     int args[]{chipIndex, nch};
-    LOG(logDEBUG1) << "Sending veto photon/file to detector [chip:" << chipIndex
-                   << ", nch:" << nch << "]";
     auto client = DetectorSocket(shm()->hostname, shm()->controlPort);
     client.Send(&fnum, sizeof(fnum));
     client.Send(args, sizeof(args));
-    client.Send(gainIndices, sizeof(int) * nch);
-    client.Send(values, sizeof(int) * nch);
+    client.Send(gainIndices.data(), sizeof(int) * nch);
```

Comment:
In case we ever would change the data of the vector this is slightly less error prone:

```cpp
client.Send(gainIndices.data(), sizeof(decltype(gainIndicies::value_type)) * nch);

```

Reply:
done, sizeof(decltype(values[0])) * nch